source/frontend/src/app/modules/expenses/components/expense-form/expense-form.component.ts

```diff
@@ -55,59 +57,79 @@ export class ExpenseFormComponent implements OnInit {
     private fb: FormBuilder,
     currency: CurrencyService,
     expensesMonthService: ExpensesMonthService,
-    private categoriesHttpClient: CategoryHttpClient,
     private expensesHttpClient: ExpensesHttpClientService,
-    private userConnectionsHttpClient: UserConnectionHttpClient) {
+    userService: UserService) {
+    this.currentUser = userService.user;
+    this.isItemShared = this.checkIfItemShared(this.item);
     this.currencies = currency.currencies;
-    this.categoriesHttpClient.getAllCategories()
+    userService.categories$
       .subscribe((categories) => this.categories = categories)
-    this.userConnectionsHttpClient.getUserConnections()
+    userService.connections$
       .subscribe((connections) => {
         this.friends = connections
           .filter((c) => c.status === UserConnectionStatus.accepted)
           .map((c) => c.person as AmbiguousUser);
 
+        if (this.item != null) {
+          // we need to add missing persons from list of permitted ones to let user select them in lookup
+          if (!this.isItemShared && this.item.permittedPersons.length > 0) {
+            this.item.permittedPersons.forEach((p) => {
+              if (!this.friends.some(f => f.id === p.id)) {
+                this.friends.push(p);
+              }
+            })
+          }
+        }
+
         if (this.item == null) {
           const lastUsers = this.lastUsersToShareExpense;
           this.form?.patchValue({
-            sharedWith: this.friends.filter((u) => lastUsers?.includes(Number(u.id)))
+            permittedPersons: this.friends.filter((u) => lastUsers?.includes(Number(u.id)))
           });
         }
       });
-    this.items$ = this.searchEntry$.pipe(
+
+    this.names$ = this.searchEntry$.pipe(
       tap(() => this.loading = true),
-      switchMap(searchEntry => this.expensesHttpClient.getExistingItems(searchEntry).pipe(
+      switchMap(searchEntry => this.expensesHttpClient.getExistingNames(searchEntry).pipe(
         catchError(() => of([])),
         tap(() => this.loading = false)
     )));
-    this.addItem = (item) => ({ item, isNew: true });
 
     this.form = this.fb.group({
       'id': [null],
       'date': [this.getCurrentDate(expensesMonthService.month), Validators.required],
-      'item': [null, Validators.required],
+      'name': [null, Validators.required],
       'priceAmount': [null, Validators.required],
       'currencyId': [this.defaultCurrency, Validators.required],
-      'category': [null],
-      'sharedWith': [null],
+      'categoryName': [null],
+      'permittedPersons': [[]],
       'description': [null]
     });
-
-    this.form.controls['date'].valueChanges
-      .subscribe(value => sessionStorage.setItem(this.lastUsedDateStorageName, JSON.stringify(value)));
   }
 
   ngOnInit(): void {
+    // populate values before saving any values to local storage
     this.populateValues(this.item);
```

Comment:
It's incorrect assuming that in `ngOnInit` we will have value in `item` variable, we need to catch updates in the variable